.github/workflows/main.yml

```diff
@@ -39,3 +39,9 @@ jobs:
               env:
                 SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
                 SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+
+            - uses: actions/upload-artifact@v4
+              if: ${{ failure() }}
+              with:
+                name: sauce-connect-log
+                path: ${{ env.SAUCE_CONNECT_DIR_IN_HOST }}/sauce-connect.log
```

Comment:
Alternatively, we could simply use the following instead:
```
path: /tmp/sauce-connect-action*/sauce-connect.log
```

That way there's no need for exporting `SAUCE_CONNECT_DIR_IN_HOST`

Reply:
Yes, but what if the implementation changes and we store the log somewhere else (not in the tmp directory), I kind of wanted to mask the implementation. Exporting SAUCE_CONNECT_DIR_IN_HOST to whatever directory we keep the log file in allows to keep the log file anywhere as long as we set some variable.  `path: /tmp/sauce-connect-action*/sauce-connect.log` would work too though, I don't have a strong opinion.